### PR TITLE
Fix multi-GPU training

### PR DESCRIPTION
**Pull request by lumurillo**
# What does this PR do?

* Change the DDP trainer's strategy.
* Remove use of trainer's `gpus` property, since it is deprecated and was affecting the use of `devices` property.
* Update training instructions on GPUs in the `README.md` file.

## Type of change

Please check all relevant options.

- [ ] Improvement (non-breaking)
- [x] Bug fix (non-breaking)
- [ ] New feature (non-breaking)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Testing

Please describe the tests that you ran to verify your changes. Consider listing any relevant details of your test configuration.

- [x] Train `FasterRCNN` model with `Carla` dataset using 1 GPU:
```
python -m mart experiment=ArmoryCarlaOverObjDet_TorchvisionFasterRCNN trainer=gpu fit=true
```
- [x] Train `FasterRCNN` model with `Carla` dataset on multiple GPUs:
```
python -m mart experiment=ArmoryCarlaOverObjDet_TorchvisionFasterRCNN trainer=ddp trainer.devices=2 fit=true
```

## Before submitting

- [x] The title is **self-explanatory** and the description **concisely** explains the PR
- [x] My **PR does only one thing**, instead of bundling different changes together
- [ ] I list all the **breaking changes** introduced by this pull request
- [x] I have commented my code
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have run pre-commit hooks with `pre-commit run -a` command without errors

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by lumurillo** [comment]
Why we need to bind `datamodel.world_size` with `trainer.devices`?

**Comment by mzweilin** [comment]
> Why we need to bind `datamodel.world_size` with `trainer.devices`?

Because `batch_size` = `datamodule.ims_per_batch` / `datamodule.world_size`

https://github.com/IntelLabs/MART/blob/6184ca764c871cc098a4e822e520f476e556a971/mart/datamodules/modular.py#L59-L65

**Comment by dxoigmn** [comment]
One could also bind `trainer.devices` to `datamodule.world_size`. Either way works, but ought to be the same number (assuming single-node multi-gpu training).

File: README.md [hunk 1]
```diff
@@ -70,6 +70,17 @@ python -m mart experiment=CIFAR10_RobustBench \
 	fit=false \
 	+trainer.limit_test_batches=1 \
 	+attack@model.modules.input_adv_test=classification_eps8_pgd10_step1
+
+# train on 1 GPU
+python -m mart experiment=ArmoryCarlaOverObjDet_TorchvisionFasterRCNN \
+	trainer=gpu \
+	fit=true
+
+# train on multiple GPUs using Distributed Data Parallel
```
Comment (mzweilin) [inline README.md:79]: Could you update with the example at https://github.com/IntelLabs/MART/pull/24#issuecomment-1351889109

**Review by mzweilin — CHANGES_REQUESTED** [review]
Would you update the two-GPU training example along with some documentation?

**Review by mzweilin — APPROVED** [review]
LGTM

**Pull request merged**